pkgs/desktops/lomiri/applications/lomiri-system-settings/wrapper.nix

```diff
@@ -0,0 +1,77 @@
+{ stdenvNoCC
+, lib
+, nixosTests
+, ayatana-indicator-datetime
+, glib
+, lndir
+, lomiri-schemas
+, lomiri-system-settings-unwrapped
+, lomiri-system-settings-security-privacy
+, maliit-keyboard
+, wrapGAppsHook
+, wrapQtAppsHook
+, plugins ? [ lomiri-system-settings-security-privacy ]
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "lomiri-system-settings";
+  inherit (lomiri-system-settings-unwrapped) version;
+
+  dontUnpack = true;
+  dontConfigure = true;
+  dontBuild = true;
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    glib # schema hook
+    lndir
+    wrapGAppsHook
+    wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    ayatana-indicator-datetime
+    glib # schema hook
+    lomiri-schemas
+    lomiri-system-settings-unwrapped
+    maliit-keyboard
+  ] ++ plugins;
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    ln -s ${lib.getExe lomiri-system-settings-unwrapped} $out/bin/${finalAttrs.meta.mainProgram}
+
+    for inheritedPath in share/lomiri-app-launch share/lomiri-url-dispatcher share/applications share/icons; do
+      mkdir -p $out/$inheritedPath
+      lndir ${lomiri-system-settings-unwrapped}/$inheritedPath $out/$inheritedPath
+    done
+
+    for mergedPath in lib/lomiri-system-settings share/lomiri-system-settings share/locale; do
+      mkdir -p $out/$mergedPath
+      for lssPart in ${lomiri-system-settings-unwrapped} ${lib.strings.concatStringsSep " " plugins}; do
+        lndir $lssPart/$mergedPath $out/$mergedPath
+      done
+    done
+
+    runHook postInstall
+  '';
+
+  dontWrapGApps = true;
+
+  preFixup = ''
+    qtWrapperArgs+=(
+      "''${gappsWrapperArgs[@]}"
+      --set NIX_LSS_PREFIX "$out"
+    )
+  '';
+
+  passthru.tests.standalone = nixosTests.lomiri-system-settings;
+
+  meta = {
+    description = "${lomiri-system-settings-unwrapped.meta.description} (wrapped)";
+    inherit (lomiri-system-settings-unwrapped.meta) homepage changelog license mainProgram maintainers platforms;
```

Comment:
```nix
  meta = lomiri-system-settings-unwrapped.meta // {
    description = "System Settings application for Lomiri (wrapped)";
    # prefer wrapper over the package
    priority = (wayfire.meta.priority or 0) - 1;
  };
```

Reply:
Hmm, I was sure that would break `meta.position` but it seems that it works fine.

Any reason to not use `description = "${lomiri-system-settings-unwrapped.meta.description} (wrapped)";`?

Reply:
1.  The possible impact is that the `Source` provided by https://search.nixos.org/packages will point to unwrapped files since they share `meta.position`.
2.  copy-pasting is for the convenience of reading. You can know the description at a glance without looking at another file.

Reply:
1: That's what I was worried about, having had such issues with some derivations. But:
```
nix-repl> lomiri.lomiri-system-settings-unwrapped.meta.position
"/home/puna/Development/nixpkgs/pkgs/desktops/lomiri/applications/lomiri-system-settings/default.nix:242"

nix-repl> lomiri.lomiri-system-settings.meta.position           
"/home/puna/Development/nixpkgs/pkgs/desktops/lomiri/applications/lomiri-system-settings/wrapper.nix:74"
```
Looks fine to me?

2: Ehh… I guess.